cpp/daal/src/algorithms/covariance/oneapi/covariance_dense_distr_step2_oneapi_impl.i

```diff
@@ -50,8 +50,13 @@ services::Status CovarianceDenseDistrStep2KernelOneAPI<algorithmFPType, method>:
 
     services::Status status;
 
+    if (crossProductTable->getNumberOfColumns() > static_cast<size_t>(services::internal::MaxVal<uint32_t>::get()))
+    {
+        return services::Status(daal::services::ErrorCovarianceInternal);
+    }
+
+    const uint32_t nFeatures    = static_cast<uint32_t>(crossProductTable->getNumberOfColumns());
```

Comment:
It looks like number of rows should be equal to a number of columns for the table. But there is no check for that.

Reply:
Size checks implemented via `DAAL_ASSERT_UNIVERSAL_BUFFER` are everywhere before each kernel call. So, the is no reason to put additional check here.